cub/thread/thread_load.cuh

```diff
@@ -416,6 +415,171 @@ __device__ __forceinline__ typename std::iterator_traits<InputIteratorT>::value_
 }
 
 
+#endif // DOXYGEN_SHOULD_SKIP_THIS
+
+
+/**
+ * \name Asynchronous Thread I/O (cache modified)
+ * @{
+ */
+
+/**
+ * \brief Thread utility for asynchronously reading memory using cub::CacheLoadModifier cache modifiers.  Can be used to load any data type.
+ *
+ * \par Example
+ * \code
+ * #include <cub/cub.cuh> // or equivalently <cub/thread/thread_load.cuh>
+ *
+ * // 4x 32-bit load using cache-always modifier:
+ * int *d_in;
+ * __shared__ int val[...];
+ * cub::ThreadLoadAsync<cub::LOAD_CA>(d_in + 0 + threadIdx.x, val + 0);
+ * cub::ThreadLoadAsync<cub::LOAD_CA>(d_in + 1 + threadIdx.x, val + 1);
+ * cub::ThreadLoadAsync<cub::LOAD_CA>(d_in + 2 + threadIdx.x, val + 2);
+ * cub::ThreadLoadAsync<cub::LOAD_CA>(d_in + 3 + threadIdx.x, val + 3);
+ * cub::ThreadLoadWait();
+
+ * // 4x 128-bit load using cache-global modifier:
+ * int4 *d_in;
+ * __shared__ int4 val[...];
+ * cub::ThreadLoadAsync<cub::LOAD_CG>(d_in + 0 + threadIdx.x, val + 0);
+ * cub::ThreadLoadAsync<cub::LOAD_CG>(d_in + 1 + threadIdx.x, val + 1);
+ * cub::ThreadLoadAsync<cub::LOAD_CG>(d_in + 2 + threadIdx.x, val + 2);
+ * cub::ThreadLoadAsync<cub::LOAD_CG>(d_in + 3 + threadIdx.x, val + 3);
+ * cub::ThreadLoadCommit();
+ * // ...
+ * cub::ThreadLoadWait();
+ * \endcode
+ *
+ * \tparam MODIFIER             <b>[inferred]</b> CacheLoadModifier enumeration
+ * \tparam InputIteratorT       <b>[inferred]</b> Input iterator type \iterator
+ */
+template <
+    CacheLoadModifier MODIFIER,
+    typename InputIteratorT>
+__device__ __forceinline__ void ThreadLoadAsync(ForwardIteratorT input, ForwardIteratorT output)
+
+
+/**
+  * \brief Establishes an ordering w.r.t previously issued ThreadLoadAsync operations.
```

Comment:
This should probably have a comment saying that this means that prior operations have been read from the source, although they have not necessarily been stored to the destination.